share/emacs/modulefiles-mode.el

```diff
@@ -0,0 +1,51 @@
+;;; modulefiles-mode.el --- Major mode for Modules modulefiles editing -*- lexical-binding: t; -*-
+
+;; Author: Laurent Besson <laurent.42.besson@gmail.com>
+;; Maintainer: Laurent Besson <laurent.42.besson@gmail.com>
+;; Created: June 12, 2025
+;; Version: 1.0
+;; Package-Requires: ((emacs "24")
+;; Keywords: Modules, modulefiles
+;; URL: https://modules.readthedocs.io/en/latest/index.html
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;; Comments:
+
+;; modulefiles-mode is a major mode for highlighting modufiles syntax
+;; it derives from tcl-mode as modulefiles is based on this language
+;; It adds colorizing for main modulefiles commands
```

Comment:
I would prefer to refer to "modulefile", in its singular form rather plural form